src/mavlink-router/endpoint.cpp

```diff
@@ -374,6 +374,13 @@ bool Endpoint::allowed_by_dropout()
     return true;
 }
 
+bool Endpoint::allowed_by_dedup(const buffer* buf)
+{
+    if (Mainloop::get_instance().add_check_dedup(buf)) {
+        return false;
+    }
+    return true;
+}
```

Comment:
Isn't this the wrong way around? `add_check_dedup()` returns true, if the packet is new and false, if it is a duplicate. So it should be allowed, when the if-condition is true.

Reply:
Good catch, thanks!